### ISSUE TITLE
Goembed does not support 1.23 due to new restrictions.

### ISSUE THREAD

**Issue by different55** [visualfc/embed#8]
As of Go 1.23, [packages are limited](https://github.com/golang/go/issues/67401) in what they're allowed to yank out of Go's internals, and that includes `go/build.parseGoEmbed` that's referenced in `parser/parser_go116.go`. Compiling this package now fails with this error:

```
link: github.com/visualfc/goembed/parser: invalid reference to go/build.parseGoEmbed
```

In the meantime, adding `-ldflags=-checklinkname=0` works as a bandaid.

**Comment by IngwiePhoenix** [comment]
Huh, now that's odd. The link name shouldn't have changed, technically.

See: https://github.com/golang/go/blob/3c05790d4847ba2d0d06b5ba3f82b311a99d531c/src/go/build/read.go#L556

The entire commit is basically an indirect "awesome-go" - except, well, you can see the comments for a giggle. I found one of my dependencies in there too...good to know that. https://github.com/golang/go/commit/9a8995b8b6a08d5fe01122771f962b36336f8aec

That said, I don't quite understand where the symbol went? Only guess I have is that I dug out an older commit and the function has either moved or be removed.

Best bet might be to attempt to outright reimplement it...?

**Comment by different55** [comment]
Yeah, the symbol's still there, but nobody's allowed to use it. There's wall there now, and I guess for particularly popular breaches of the wall are they letting anything through with that go:linkname directive.

I think you're right that the best move is to just duplicate the implementation into this package.

**Comment by visualfc** [comment]
fix https://github.com/visualfc/goembed/pull/10
